### PR TITLE
Issue #4950: File usage has bad link

### LINKED ISSUE #4950: File usage has bad link.
**Description of the bug**

After uploading a fie to a Backdrop site, the file shows up in admin/content/files with some links to manage it.  Clicking "Manage" gives the user three tabs, including "Manage" and "Usage".  On the "Usage" page is a link to the file itself, but the link is malformed.

**Steps To Reproduce**

To reproduce the behavior:
1. Go to admin/content/files, click "Add a file...", and add something.
2. Go back to admin/content/files and click "Manage" for the file you just uploaded.
3. Click the "Usage" tab
4. Inspect the link to the file.

**Actual behavior**

The link is to something like  "http://example.com/public://myfile.txt".  Note the "public://" in the middle of the URL.

**Expected behavior**

The link should be to "http://example.com/files/myfile.txt." 

In `file_usage_page()`, the path needs to be translated to a URL with `file_create_url()` before being passed to `l()`.

**Additional information**

Add any other information that could help, such as:
- Backdrop 1.18.1

### PR DESCRIPTION
**Pull request by indigoxela**
Fixes https://github.com/backdrop/backdrop-issues/issues/4950

### CODE REVIEW

File: core/modules/file/file.pages.inc [hunk 1]
```diff
@@ -1023,7 +1023,16 @@ function file_usage_page($file) {
 
         // Link the label to the URI when possible.
         if (!empty($entity_uri['path'])) {
-          $entity_label = l($label, $entity_uri['path']);
+          // Unlike Node or Term entities File entities have a stream wrapper
+          // to the file set as their path, but the path to the entities' page
+          // is "path/[fid]".
+          if ($entity_type == 'file') {
+            $uri = "file/{$entity_id}";
```
Comment (klonos) [inline core/modules/file/file.pages.inc]: Excuse my ignorance, but why the curly syntax here?

Comment (indigoxela) [reply core/modules/file/file.pages.inc]: In fact, it's very unlikely here that php mixes up the string with the variable, so the curly braces probably aren't necessary.
But they don't do any harm, either.

Do you insist in removing them?

Comment (jlfranklin) [reply core/modules/file/file.pages.inc]: The typical code style in Backdrop for something like this is concatenation, not interpolation

```php
$uri = 'file/' . $entity_id;
```

Comment (indigoxela) [reply core/modules/file/file.pages.inc]: > The typical code style in Backdrop for something like this...

Not sure, but why not.

File: core/modules/file/file.pages.inc [hunk 2]
```diff
@@ -1023,7 +1023,16 @@ function file_usage_page($file) {
 
         // Link the label to the URI when possible.
         if (!empty($entity_uri['path'])) {
-          $entity_label = l($label, $entity_uri['path']);
+          // Unlike Node or Term entities File entities have a stream wrapper
+          // to the file set as their path, but the path to the entities' page
+          // is "path/[fid]".
+          if ($entity_type == 'file') {
```
Comment (quicksketch) [inline core/modules/file/file.pages.inc]:
```suggestion
          // is "path/[fid]".
          // @todo Fix in https://github.com/backdrop/backdrop-issues/issues/2300
          if ($entity_type == 'file') {
```

**Comment by quicksketch** [comment]
I added a `@todo` line in the code to reference fixing this in https://github.com/backdrop/backdrop-issues/issues/2300. Merged into 1.x and 1.18.x. Thanks!

**Pull request merged**